serde-generate/src/java.rs

```diff
@@ -415,7 +415,9 @@ serializer.sort_map_entries(offsets);
                 write!(
                     self.out,
                     r#"
-assert value.length == {};
+if (value.length != {}) {{
+    throw new java.lang.IllegalArgumentException("Invalid length for fixed-size array");
```

Comment:
"Invalid length(" + value.length + ") for fixed-size ({})...", add the size to make it specific error message